lib/matplotlib/axes/_axes.py

```diff
@@ -4147,6 +4147,11 @@ def bxp(self, bxpstats, positions=None, widths=None, vert=True,
         --------
         .. plot:: gallery/statistics/bxp.py
         """
+        # Clamp median line to edge of box by default.
+        medianprops = {
+            "solid_capstyle": "butt",
```

Comment:
I think we should unversally "solid_capstyle" and "dashed_capstyle" to "butt" for both `medianprops` and `meanprops`. We want the lines to end within the box for both independent of the global capstyle settings.

Reply:
will do